ntp-proto/src/packet.rs

```diff
@@ -134,6 +138,8 @@ impl NtpHeader {
     }
 
     pub fn serialize(&self) -> [u8; 48] {
+        // Version should only ever be set internally in this module, so
+        // violations of this should never happen.
```

Comment:
I'd make the assert a `debug_assert!()` here, we probably wouldn't want to blow up a production deployment?

Reply:
Given what NTP does, especially on a client I think the safer option is to actually blow up in case of a programming error and just not adjust the clock anymore, rather than ignore a potential issue and in turn maybe start steering the clock in the wrong direction, as in that case the failure mode is much more subtle. However, I think this is a discussion we should separate from the actual code, so I have created issue #24 for that.